cockatrice/src/player.cpp

```diff
@@ -1972,6 +1975,45 @@ bool Player::clearCardsToDelete()
     return true;
 }
 
+void Player::actMoveCardXCardsFromTop()
+{
+    bool ok;
+    int number = QInputDialog::getInt(0, tr("Place card X cards below top of library"), tr("Number of cards to place card below:"), defaultNumberTopCardsToPlaceBelow, 0, 2000000000, 1, &ok);
+    if (ok)
+        defaultNumberTopCardsToPlaceBelow = number;
+
+    QList<QGraphicsItem *> sel = scene()->selectedItems();
+    QList<CardItem *> cardList;
+    while (!sel.isEmpty())
+        cardList.append(qgraphicsitem_cast<CardItem *>(sel.takeFirst()));
+
+    QList< const ::google::protobuf::Message * > commandList;
+    ListOfCardsToMove idList;
+    for (int i = 0; i < cardList.size(); ++i)
+        idList.add_card()->set_card_id(cardList[i]->getId());
+
+    if (cardList.isEmpty())
+        return;
+
+    int startPlayerId = cardList[0]->getZone()->getPlayer()->getId();
+    QString startZone = cardList[0]->getZone()->getName();
+
+    Command_MoveCard *cmd = new Command_MoveCard;
+    cmd->set_start_player_id(startPlayerId);
+    cmd->set_start_zone(startZone.toStdString());
+    cmd->mutable_cards_to_move()->CopyFrom(idList);
+    cmd->set_target_player_id(getId());
+    cmd->set_target_zone("deck");
+    cmd->set_x(number);
```

Comment:
What happens if I set `number = 100` in the UI but I only have 60 cards in the deck?